From 1d1e7a17274f894d8db1a9b44fc5fe14949fbacc Mon Sep 17 00:00:00 2001
From: Andrew Kelley <andrew@ziglang.org>
Date: Wed, 3 Sep 2025 16:35:02 -0700
Subject: [PATCH 106/133] Sema: fix accessing ptr field of double array pointer
 with sentinel
X-Developer-Signature: v=1; a=openpgp-sha256; l=854; i=xtex@astrafall.org;
 h=from:subject; bh=VPHdgIDLLWHuTnP+MKJS2XJ/5euUb8XeYjhEYW968rQ=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78rjtL3ztfWO9OaWP9+m6Os6Le++pyKeeYHUpkZ6z+
 NA/n2PXOyayMIhxMViKKbIUGTZ4s+qk84suK5eFmcPKBDJEWqSBgYGBgYWBLzcxr9RIx0jPVNtQ
 z9BQx0jHiIGLUwCm+o8CI8MOFZbmPeY9XaW6Oo/yLCfLnyhJirV6d7KUiytDZ8siJS6GP9yGYkI
 iJZypAeYvvMzt2LZlc1xJNp117aaV8cbkpS4fWAA=
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

(cherry picked from commit 772793c004e387bd37c0cd3b7f849936e7fdae4c)
---
 src/Sema.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Sema.zig b/src/Sema.zig
index 5c89060f02df..3078fb344a48 100644
--- a/src/Sema.zig
+++ b/src/Sema.zig
@@ -26918,7 +26918,7 @@ fn fieldPtr(
                 const ptr_info = object_ty.ptrInfo(zcu);
                 const new_ptr_ty = try pt.ptrTypeSema(.{
                     .child = Type.fromInterned(ptr_info.child).childType(zcu).toIntern(),
-                    .sentinel = if (object_ty.sentinel(zcu)) |s| s.toIntern() else .none,
+                    .sentinel = if (inner_ty.sentinel(zcu)) |s| s.toIntern() else .none,
                     .flags = .{
                         .size = .many,
                         .alignment = ptr_info.flags.alignment,
-- 
2.52.0

